src/app/theme-provider.tsx

```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import { ThemeProvider as NextThemesProvider } from 'next-themes';
+import { ThemeProviderProps } from 'next-themes/dist/types';
+import { useState, useEffect } from 'react';
+
+export const ThemeProvider = ({ children, ...props }: ThemeProviderProps) => {
```

Comment:
Let's talk about this I dont think this is good idea and I am being soft :D

Reply:
Hey, it really seems to be fine look here: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props:~:text=%7D-,Supported%20Pattern%3A%20Passing%20Server%20Components%20to%20Client%20Components%20as%20Props,the%20children%20prop.%20You%20can%20use%20any%20prop%20to%20pass%20JSX.,-Previous  - you have servers component that is /app/layout and than you will import client component - you can still pass server components as a prop or child inside this provider...

Reply:
By the way,  `<AuthProvider>` also uses "use client" inside already and wraps the `<Layout>` 😅

Reply:
And I also made a few tests 
<img width="774" alt="image" src="https://github.com/GLObus303/sportujspolu/assets/47262567/4cfb4242-e607-4386-90b4-c000ac3029a1">

<img width="310" alt="image" src="https://github.com/GLObus303/sportujspolu/assets/47262567/3a60afb0-960e-407a-b397-546461799fda">

1) console.log logs into VSCode terminal and not browser so it was on the server side. 
2) use react hook inside server component - and still I get server components error:

<img width="873" alt="image" src="https://github.com/GLObus303/sportujspolu/assets/47262567/5d17a847-7ad8-4d81-a8d4-9e16778378f7">


```
./src/app/page.tsx
ReactServerComponentsError:

You're importing a component that needs useEffect. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
Learn more: https://nextjs.org/docs/getting-started/react-essentials

   ╭─[/Users/jo/sportujspolu/src/app/page.tsx:1:1]
 1 │ import { useEffect } from 'react';
   ·          ─────────
 2 │ 
 3 │ import { Events } from '../components/Events';
 4 │ import { getAllEvents } from '../api/events';
   ╰────

Maybe one of these should be marked as a client entry with "use client":
  ./src/app/page.tsx
```